Core/Src/app_threadx.c

```diff
@@ -213,21 +210,29 @@ void thread_0_entry(ULONG thread_input)
 
 void thread_1_entry(ULONG thread_input)
 {
+	ULONG received_message;
+	ULONG queue_status;
+	ULONG semaphore_status;
+	char message_buffer[100]; // Buffer to store the message string
 
 	HAL_UART_Transmit(&huart2, (uint8_t*)"Made Thread 1.\r\n", 17, 100);
-	char message_buffer[100]; // Buffer to store the message string
-    /* This thread simply sits in while-forever-sleep loop.  */
+
+	/* Makes sure that thread 2 is create before starting the while loop. */
+	tx_semaphore_get(&semaphore_0, TX_WAIT_FOREVER);
```

Comment:
Since when the threads are first created thread 2 tries to receive a number from the queue after thread 1 took the only number in it. This causes thread 2 to be suspended and thread 0 and thread 1 to run again causing thread 2 to be out of sync from the other 2 threads. I did a lab that dealt with this issue in my RTOS class and was told that it was better to use a semaphore like this to synchronize the threads instead of doing something like starting thread 0 with sending a 0 to the queue to keep thread 2 from suspending.

Reply:
I see, can you document that?